### ISSUE TITLE
Enumerable helpers

### ISSUE THREAD

**Issue by Igorbek** [yuriyoleynikov/YoTennis#21]
This task is to implement the following helpers for `Enumerable`:

```cs
internal static class EnumerableExtensions
{
    public static IEnumerable<T> WithElement<T>(this IEnumerable<T> source, T element) { ... }
    public static IEnumerable<T> WithoutElement<T>(this IEnumerable<T> source, T element, IEqualityComparer<T> comparer = null) { ... }
}
```

- `WithElement` must return sequence with added passed element
- `WithoutElement` must return sequence with added passed element
- both methods allow to accept `null` as a `source`, that means the collection is empty
- they must return non-null always
- implementation must not use `List` or `Array` or anything buffering. **use `yield return`/`yield break`**
- [**new**] method accept optional _equality comparer_ of type `IEqualityComparer<T>`
  - if `comparer` is not specified or `null` passed, use default comparer from `EqualityComparer.Default`

**Comment by Igorbek** [comment]
Usage:
```cs
IEnumerable<int> seq = ...
IEnumerable<int> seq = seq.WithElement(10).WithoutElement(1);
```

**Comment by Igorbek** [comment]
@yuriyoleynikov task has been updated

**Comment by Igorbek** [comment]
implemented in #20
